WTG.Analyzers/Analyzers/Regex/RegexCodeFixProvider.cs

```diff
@@ -0,0 +1,216 @@
+				}
+				else if (newInner.HasAnnotation(Discardable))
+				{
+					return NoneOptions.WithTriviaFrom(node);
+				}
+
+				return node
+					.WithExpression((ExpressionSyntax)newInner)
+					.WithAdditionalAnnotations(Simplifier.Annotation);
+			}
+
+			public override SyntaxNode VisitBinaryExpression(BinaryExpressionSyntax node)
+			{
+				return (node.Kind()) switch
+				{
+					SyntaxKind.BitwiseOrExpression => VisitBitwiseOr(node),
+					SyntaxKind.BitwiseAndExpression => VisitBitwiseAnd(node),
+					_ => Visit(node),
+				};
+			}
+
+			SyntaxNode VisitBitwiseOr(BinaryExpressionSyntax node)
+			{
+				var diff = Compare(node.OperatorToken.Span, syntax.FullSpan);
+
+				if (diff > 0)
+				{
+					var tmp = Visit(node.Left);
+
+					if (tmp.HasAnnotation(Discardable))
+					{
+						return node.Right.WithTriviaFrom(node);
+					}
+
+					return node.WithRight((ExpressionSyntax)tmp);
+				}
+				else if (diff < 0)
+				{
+					var tmp = Visit(node.Right);
+
+					if (tmp.HasAnnotation(Discardable))
+					{
+						return node.Left.WithTriviaFrom(node);
+					}
+
+					return node.WithLeft((ExpressionSyntax)tmp);
+				}
+
+				return node;
+			}
+
+			SyntaxNode VisitBitwiseAnd(BinaryExpressionSyntax node)
+			{
+				var diff = Compare(node.OperatorToken.Span, syntax.FullSpan);
+
+				if (diff > 0)
+				{
+					var left = Visit(node.Left);
+					return left.HasAnnotation(Discardable)
+						? NoneOptions.WithTriviaFrom(node)
+						: node.WithLeft((ExpressionSyntax)left);
+				}
+				else if (diff < 0)
+				{
+					var right = Visit(node.Right);
+					return right.HasAnnotation(Discardable)
+						? NoneOptions.WithTriviaFrom(node)
+						: node.WithRight((ExpressionSyntax)right);
+				}
+
+				return node;
+			}
+
+			// returns:
+			// -1: if x appears entirly before y.
+			//  0: if x and y overlap.
+			//  1: if x appears entirely after y.
+			static int Compare(TextSpan x, TextSpan y)
+			{
+				if (x.End <= y.Start)
+				{
+					return -1;
+				}
+				else if (y.End <= x.Start)
+				{
+					return 1;
+				}
+				else
+				{
+					return 0;
+				}
+			}
+
+			static readonly SyntaxAnnotation Discardable = new SyntaxAnnotation(nameof(Discardable));
+
+			static readonly ExpressionSyntax NoneOptions =
+				SyntaxFactory.MemberAccessExpression(
+					SyntaxKind.SimpleMemberAccessExpression,
+					SyntaxFactory.MemberAccessExpression(
+						SyntaxKind.SimpleMemberAccessExpression,
+						SyntaxFactory.MemberAccessExpression(
+							SyntaxKind.SimpleMemberAccessExpression,
+							SyntaxFactory.MemberAccessExpression(
+								SyntaxKind.SimpleMemberAccessExpression,
+								SyntaxFactory.IdentifierName("System"),
+								SyntaxFactory.IdentifierName("Text")),
+							SyntaxFactory.IdentifierName("RegularExpressions")),
+						SyntaxFactory.IdentifierName(nameof(RegexOptions))),
+					SyntaxFactory.IdentifierName(nameof(RegexOptions.None)))
+				.WithAdditionalAnnotations(Simplifier.Annotation)
+				.WithAdditionalAnnotations(Discardable);
```

Comment:
we add something that we expect to remove later?

Reply:
I assume you mean the `Discardable` annotation. This is just so that a Visit method can distinguish between "'None' because we replaced something" and "'None' because it was already there". The former may be folded into another thing in some contexts.

Eg.
```csharp
RegexOptions.Compiled | RegexOptions.IgnoreCase
```
becomes
```csharp
RegexOptions.IgnoreCase
```
rather than
```csharp
RegexOptions.None | RegexOptions.IgnoreCase
```